tests/kafkatest/services/kafka.py

```diff
@@ -199,29 +208,53 @@ def restart_node(self, node, wait_sec=0, clean_shutdown=True):
     def leader(self, topic, partition=0):
         """ Get the leader replica for the given topic and partition.
         """
-        cmd = "/opt/kafka/bin/kafka-run-class.sh kafka.tools.ZooKeeperMainWrapper -server %s " \
-              % self.zk.connect_setting()
-        cmd += "get /brokers/topics/%s/partitions/%d/state" % (topic, partition)
-        self.logger.debug(cmd)
+        topic_partition_data = self.zk_get_data("/brokers/topics/%s/partitions/%d/state" % (topic, partition))
+
+        if topic_partition_data is None:
+            raise Exception("Error finding data for topic %s and partition %d." % (topic, partition))
+        self.logger.info(topic_partition_data)
+
+        leader_idx = int(topic_partition_data["leader"])
+        self.logger.info("Leader for topic %s and partition %d is now: %d" % (topic, partition, leader_idx))
+        return self.get_node(leader_idx)
+
+    def controller(self):
+        """
+        Get the current controller
+        """
+        controller_data = self.zk_get_data("/controller")
+        if controller_data is None:
+            raise Exception("Error finding controller.")
+        self.logger.info(controller_data)
 
+        controller_idx = int(controller_data["brokerid"])
+        self.logger.info("Controller is: %d" % controller_idx)
+        controller_node = self.get_node(controller_idx)
+
+        if controller_node is None:
+            raise Exception("Found no node corresponding to the id: %d" % controller_idx)
+        return controller_node
+
+    def zk_get_data(self, path):
```

Comment:
Should we just move this to the Zookeeper service? Seems like it's a better home for it.